crates/bevy_utils/src/debug_info.rs

```diff
@@ -132,6 +132,20 @@ cfg::alloc! {
             Self::owned(value)
         }
     }
+
+    impl From<DebugName> for Cow<'static, str> {
+        #[cfg_attr(
+            not(feature = "debug"),
+            expect(
+                unused_variables,
+                reason = "The value will be ignored if the `debug` feature is not enabled"
+            )
+        )]
+        fn from(value: DebugName) -> Self {
+            #[cfg(feature = "debug")]
+            value.name
+        }
```

Comment:
What does this return when `debug` is not enabled? Looks like a compile error to me